compiler/rustc_const_eval/src/interpret/eval_context.rs

```diff
@@ -712,10 +714,19 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         let mut locals = IndexVec::from_elem(dummy, &body.local_decls);
 
         // Now mark those locals as live that have no `Storage*` annotations.
+        // Note that this is a dicey situation: some of these locals can be unsized function parameters,
+        // and immediates are not allowed to be unsized! When pushing a stack frame with such unsized
+        // locals, one must immediately allocate proper amounts of memory for those locals.
         let always_live = always_storage_live_locals(self.body());
         for local in locals.indices() {
             if always_live.contains(local) {
-                locals[local].value = LocalValue::Live(Operand::Immediate(Immediate::Uninit));
+                let ty = body.local_decls[local].ty;
+                let ty = self.subst_from_current_frame_and_normalize_erasing_regions(ty)?;
+                if ty.is_sized(*self.tcx, self.param_env) {
```

Comment:
I'm worried this will be too expensive to do for each local. But how else can we avoid initializing unsized locals to `Immediate::Uninit`, which is just not a valid state for the interpreter to be in?

Reply:
Ah, this already doesn't work. ConstProp says TooGeneric.